extensions/sql-migration/src/dialog/assessmentResults/sqlDatabasesTree.ts

```diff
@@ -116,105 +102,35 @@ export class SqlDatabaseTree extends AssessmentDialogComponent {
 				dataValues: [
 				]
 			}
-		);
-
-		let dbList = await azdata.connection.listDatabases(this._model.sourceConnectionId);
-
-		if (dbList.length > 0) {
-			let rowNumber = 0;
-			this._assessmentData.forEach((value, key) => {
-				this.databaseTable.component().dataValues?.push(
-					[
-						{
-							value: false,
-							style: styleLeft
-						},
-						{
-							value: key,
-							style: styleLeft
-						},
-						{
-							value: value.length,
-							style: styleRight
-						}
-					]
-
-				);
-				let dbIssues = {
-					name: key,
-					issues: value
-				};
-				mapRowIssue.set(rowNumber, dbIssues);
-				dbList = dbList.filter(obj => obj !== key);
-
-				rowNumber = rowNumber + 1;
-			});
-
-			dbList.filter(db => !SqlDatabaseTree.excludeDbs.includes(db)).forEach((value) => {
-				this.databaseTable.component().dataValues?.push(
-					[
-						{
-							value: true,
-							style: styleLeft
-						},
-						{
-							value: value,
-							style: styleLeft
-						},
-						{
-							value: 0,
-							style: styleRight
-						}
-					]
-
-				);
-				let impactedObjects: SqlMigrationImpactedObjectInfo[] = [];
-				let issue: Issues[] = [{
-					description: 'No Issues',
-					recommendation: 'No Issues',
-					moreInfo: 'No Issues',
-					impactedObjects: impactedObjects,
-					rowNumber: rowNumber
-				}];
-				let noIssues = {
-					name: value,
-					issues: issue
-				};
-				mapRowIssue.set(rowNumber, noIssues);
-				rowNumber = rowNumber + 1;
-			});
-		}
-
-		this.databaseTable.component().onRowSelected(({ row }) => {
-			const rowInfo = mapRowIssue.get(row);
-			if (rowInfo) {
-				this._assessmentResultsTable.component().dataValues = [];
-				this._dbName.value = rowInfo.name;
-				if (rowInfo.issues[0].description === 'No Issues') {
-					this._recommendation.value = `Warnings (0 issues found)`;
-				} else {
-					this._recommendation.value = `Warnings (${rowInfo.issues.length} issues found)`;
-				}
-
-				// Need some kind of refresh method for declarative tables
-				let dataValues: string[][] = [];
-				rowInfo.issues.forEach(async (issue) => {
-					dataValues.push([
-						issue.description
-					]);
-
-				});
-
-				this._assessmentResultsTable.component().updateProperties({
-					data: dataValues
-				});
-
-			}
+		).component();
 
+		this._model._assessmentResults.databaseAssessments.forEach((db) => {
+			this._databaseTable.dataValues?.push(
```

Comment:
Instead of adding these one by one you should make the array of them and then set it at once so that you avoid having a bunch of unnecessary calls going over the JRPC layer